src/sidebar/store/modules/activity.ts

```diff
@@ -2,51 +2,46 @@
  * Store module which tracks activity happening in the application that may
  * need to be reflected in the UI.
  */
+import type { Annotation } from '../../../types/api';
 import { createStoreModule, makeAction } from '../create-store';
 
-/** @typedef {import('../../../types/api').Annotation} Annotation */
-
-const initialState = {
+export type State = {
   /**
    * Annotation `$tag`s that correspond to annotations with active API requests
-   *
-   * @type {string[]}
    */
-  activeAnnotationSaveRequests: [],
+  activeAnnotationSaveRequests: string[];
+
+  /** The number of API requests that have started and not yet completed. */
+  activeApiRequests: number;
+  /** The number of annotation fetches that have started and not yet completed. */
+  activeAnnotationFetches: number;
+  /** Have annotations ever been fetched? */
+  hasFetchedAnnotations: boolean;
+
   /**
-   * The number of API requests that have started and not yet completed.
+   * The number of total annotation results the service reported as
+   * matching the most recent load/search request
    */
+  annotationResultCount: number | null;
+};
```

Comment:
I have chosen to manually define the `State` type and set it in `initialState`, instead of inferring it as we were doing.

The reason is that there are things that need some kind of type annotation anyway (an empty array would be inferred as `unknown[]`, for example).

In the `annotation` module we used a different approach, which is defining the type of `initialState` with an in-place casting, like this:

```ts
const initialState = {
  // ...
} as {
  // Type definition goes here...
}

export type Ssate = typeof initialState;
```

But I see two main problems there:

* We could accidentally define props in `initialState` that are not part of the type defined after `as`, and the transpiler would not complain, making it potentially fail later at runtime.
* We are explicitly defining the type anyway, after `as`, so we may as well define a regular type and use it.

Reply:
Explicitly defining the state is good with me.